lib/deprecation_tracker.rb

```diff
@@ -74,9 +74,15 @@ def self.init_tracker(opts = {})
     transform_message = opts[:transform_message]
     deprecation_tracker = DeprecationTracker.new(shitlist_path, transform_message, mode)
     if defined?(ActiveSupport)
-      ActiveSupport::Deprecation.behavior << -> (message, _callstack = nil, _deprecation_horizon = nil, _gem_name = nil) {
-        deprecation_tracker.add(message)
-      }
+      if Rails::VERSION::STRING < '5.0'
```

Comment:
From my tests for this blog post https://github.com/ombulabs/fastruby.io/pull/625, the 4 arguments is a thing since Rails 5.2, not 5.0.

Is this string comparison for the version working as expected in general? maybe for Rails versions it's enough since it never gets to `X.10`?

we use the Gem::Version class in other places to be sure it's comparing versions properly https://github.com/fastruby/next_rails/blob/main/lib/deprecation_tracker.rb#L26C8-L26C68

Reply:
I ran the deprecation tracker in Rails 5 as it was (with the 4 arguments), and it worked just fine.
Will update it to use the `Gem::Version`

Reply:
I'm confused, because I see in the rails 5.0 source that the behaviors should have 2 arguments https://github.com/rails/rails/blob/c4d3e202e10ae627b3b9c34498afb45450652421/activesupport/lib/active_support/deprecation/behaviors.rb#L11

Reply:
@arielj, what about if we rescue the `ArgumentError`, that way we won't specify a Rails version 

```
  ActiveSupport::Deprecation.behavior << -> (message, _callstack = nil, _deprecation_horizon = nil, _gem_name = nil) {
    deprecation_tracker.add(message)
  }
rescue
  ActiveSupport::Deprecation.behavior << -> (message, _callstack) {
    deprecation_tracker.add(message)
  }